src/content/reference/react/use.md

```diff
@@ -269,13 +290,130 @@ export function fetchMessage() {
+
+import { use, Suspense, useState } from 'react';
+import { fetchByline } from './api.js';
+
+function Article({ title, body, shouldIncludeByline, authorId }) {
+  const [byline, setByline] = useState(null);
+
+  if (shouldIncludeByline && !byline) {
+    // Because `use` is inside a conditional block, we avoid blocking
+    // unnecessarily when `shouldIncludeByline` is false.
+    setByline(use(fetchByline(authorId)));
+  }
+
+  return (
+    <div>
+      <h1>{title}</h1>
+      <h2>{byline}</h2>
+      <section>{body}</section>
+    </div>
+  );
+}
+
+export default function App() {
+  const [show, setShow] = useState(false);
+  function showArticle() {
+    setShow(true);
+  }
+
+  if (!show) {
+    return <button onClick={showArticle}>Show article</button>;
+  }
+  return (
+    <Suspense fallback={<p>⌛Loading article...</p>}>
+      <Article
+        title={'Article Title'}
+        body={'this is the body of an article'}
+        authorId={'Jordan'}
+        shouldIncludeByline={true}
+      />
+    </Suspense>
+  );
+}
+```
+
+```js index.js hidden
+// TODO: update to import from stable
+// react instead of canary once the `use`
+// Hook is in a stable release of React
+import React, { StrictMode } from 'react';
+import { createRoot } from 'react-dom/client';
+import './styles.css';
+
+// TODO: update this example to use
+// the Codesandbox server component
+// demo environment once it is created
+import App from './App';
+
+const root = createRoot(document.getElementById('root'));
+root.render(
+  <StrictMode>
+    <App />
+  </StrictMode>
+);
+```
 
+```js api.js hidden
+export function fetchByline(authorId) {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve('By ' + authorId);
+    }, 200);
+  });
+}
+```
+
+```css styles.css hidden
+* {
+  box-sizing: border-box;
+}
 
+body {
+  font-family: sans-serif;
+  margin: 20px;
+  padding: 0;
+}
+
+h1 {
+  margin-top: 0;
+  font-size: 22px;
+}
+
+h2 {
+  margin-top: 0;
+  font-size: 15px;
+}
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "18.3.0-canary-9377e1010-20230712",
+    "react-dom": "18.3.0-canary-9377e1010-20230712",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js"
+}
+```
+</Sandpack>
+
+<Solution />
+
+</Recipes>
 
 ### Reading context with `use` {/*reading-context-with-use*/}
 
-When a [context](/learn/passing-data-deeply-with-context) is passed to `use`, it works similarly to [`useContext`](/reference/react/useContext). While `useContext` must be called at the top level of your component, `use` can be called inside conditionals like `if` and loops like `for`.
+When a [context](/learn/passing-data-deeply-with-context) is passed to `use`, it works similarly to [`useContext`](/reference/react/useContext). While `useContext` must be called at the top level of your component, `use` can be called inside conditionals like `if` and loops like `for`. `use` is preferred over `useContext` because it is more flexible.
```

Comment:
Does this mean that people generally shouldn't be using `useContext` anymore? That's how I'd interpret this guidance

Reply:
Yes, I  added this afer @sophiebits suggested we answer the question "is there a reason to use useContext over use?" I don't think there is, so I added this sentence to indicate that you should use `use` over `useContext`: https://github.com/reactjs/react.dev/pull/6177#discussion_r1281183548